### ISSUE TITLE
TSQL L027 not working scenario

### ISSUE THREAD

**Issue by tkkasimov** [sqlfluff/sqlfluff#3592]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The rule doesn't return any violation when a column is specified in WHERE clause without a table reference, while there is expression in SELECT clause that has the same name as the column name

### Expected Behaviour

For other dialects it returns the following result:
_Unqualified reference 'a' found in select with more than one referenced table/view._

### Observed Behaviour

No violation in output

### How to reproduce

```
select 
     a = 1, 
     t2.b 
from table1 t1 
join table2 t2 on t1.x = t2.x 
where a = 2
```

### Dialect

tsql

### Version

1.1.0

### Configuration

[sqlfluff]
dialect = tsql

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
T-SQL allows the alternative way os specifying aliases (`a = 1`) that other dialects do not.

The equivalent for other dialects would be:

```sql
select 
     1 as a, 
     t2.b 
from table1 t1 
join table2 t2 on t1.x = t2.x 
where a = 2
```

In both cases `a` is a valid alias, and not a column reference so does not need to be fully qualified.

However, if you use `a = 1` in a non-tsql dialect it is not an alias, but an expression and so `a` is a column reference - hence why it flags.

If it is both an alias and a column reference in tsql then not sure what tsql will use it as (and similarly if you use `1 as a` in another dialect and a column `a` exists in one of the tables), but think that's beyond what SQLFluff can reasonably be expected to check for since it doesn't know the underlying table structure to know if a column `a` exists.

**Comment by tunetheweb** [comment]
Closing this as explanation given above.
